### PR TITLE
#24: google auth 

### LINKED ISSUE #24: implement passport.js google oauth

### PR DESCRIPTION
**Pull request by atakann**
- add google auth template in middleware
- add user routes into app.js

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -2,4 +2,8 @@ NODE_LOCAL_PORT=3000
 MONGODB_URI='mongodb://127.0.0.1:27017'
 DB_NAME= 'volunet'
 
-DEBUG = app:*
\ No newline at end of file
+DEBUG = app:*
+
+GOOGLE_CLIENT_ID=686293049369-neqjv8uibtdmjik1acmc7v616t67vgg0.apps.googleusercontent.com
+GOOGLE_CLIENT_SECRET=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
+GOOGLE_CALLBACK_URL=http://localhost:3000/oauth2/redirect/google
```
Comment (Younesnz) [inline .env]: These are secret! they shouldn't be on a public repo on GitHub! someone can use them.

File: src/middleware/googleAuth.js
Comment (Younesnz) [inline src/middleware/googleAuth.js:1]: where is this middleware used? should there be any exports?

**Review by Younesnz — CHANGES_REQUESTED** [review]

File: .env (hunk 1 above)
Comment (atakann) [reply .env]: Even though this was in .gitignore, apparently it was still being tracked. So "git rm .env --cached" is used to untrack it, recommend it for everyone.

File: src/middleware/googleAuth.js
Comment (atakann) [reply src/middleware/googleAuth.js:1]: its being used in app.js, let this be a template for google auth using passport.js. we build on top of it after we get the login page and login with google button

**Review by abdulrahmanalbakkar — APPROVED** [review]

**Review by sheidanouri — APPROVED** [review]

**Review by Younesnz — APPROVED** [review]

**Pull request merged**